Tella/Data/VaultManager/VaultFilesManager.swift

```diff
@@ -77,45 +85,48 @@ class VaultFilesManager :ObservableObject, VaultFilesManagerInterface {
         
         return subject.eraseToAnyPublisher()
     }
-
+    
     func addVaultFile(fileDetail:VaultFileDetails,filePath: URL, parentId: String?, deleteOriginal:Bool) -> AnyPublisher<BackgroundActivityStatus,Never> {
         
         let subject = CurrentValueSubject<BackgroundActivityStatus, Never>(.inProgress)
-
-        if let  isSaved = self.vaultManager?.save(filePath, vaultFileId: fileDetail.file.id) {
-
-            if isSaved {
+        Task {
+            guard let filePath = await getModifiedURL(importedFile: fileDetail.importedFile) else { subject.send(.failed)
+                return}
+            
+            if let  isSaved = self.vaultManager?.save(filePath, vaultFileId: fileDetail.file.id) {
                 
-                 let result = self.vaultDataBase.addVaultFile(file: fileDetail.file, parentId: parentId)
-               
-                switch result {
-                case .success:
-                    guard let vaultFile = getVaultFile(id: fileDetail.file.id) else {
+                if isSaved {
+                    
+                    let result = self.vaultDataBase.addVaultFile(file: fileDetail.file, parentId: parentId)
+                    
+                    switch result {
+                    case .success:
+                        guard let vaultFile = getVaultFile(id: fileDetail.file.id) else {
+                            subject.send(.failed)
+                            return
+                        }
+                        
+                        shouldReloadFiles.send(true)
+                        
+                        // Delete original file
+                        handleDeletionFiles(importedFiles:[fileDetail.importedFile], deleteOriginal: deleteOriginal)
+                        
+                        subject.send(.completed(vaultFile))
+                        
+                    default:
                         subject.send(.failed)
```

Comment:
maybe using failure instead defaullt making more sens to send .failed?

Reply:
` .failed ` is a `BackgroundActivityStatus`, maybe send a `Fail`
How can we modified with failure

Reply:
I meant replace 

> `default: ` by `case .failure:`